core/state_transition_pmh.go

```diff
@@ -86,6 +87,10 @@ func (pmh *privateMessageHandler) verify(vmerr error) (bool, error) {
 	log.Trace("Verify hashes of affected contracts", "expectedHashes", pmh.receivedPrivacyMetadata.ACHashes, "numberOfAffectedAddresses", len(actualACAddresses))
 	privacyFlag := pmh.receivedPrivacyMetadata.PrivacyFlag
 	for _, addr := range actualACAddresses {
+		if core.ContractWhitelistMap != nil && core.ContractWhitelistMap.GetContractWhitelistByAddress(addr) {
+			// introduce a whitelist to allow whitelisting of contracts for EP simulation skips
+			continue
+		}
```

Comment:
It appears that few validations are being skipped. Should these be moved to before line 120 instead - so that other validations will still execute? Let me know.

Reply:
good catch, amended

Reply:
we actually _want_ to skip those validations as well. For affected contracts
- privacyMetaData exists
  - this check is used as precedence for validity of next check. Also as a check to see if this is a private contract address, and if contract is deployed/extended to the current node.
- privacyFlagMismatch
  - this is to ensure that privacyFlag is the same, to protect enhanced privacy functionality

The intention of partior whitelist is to mark contracts as 'safeToInvoke' as these are deployed by partior and we have vetted that they are private contracts with the correct privacy flag. 

Enforcing these checks will allow for transactions to fail for undeployed contracts and break the current dummy calls implementation when SU privacy is fixed.